src/components/infinite-scroll/InfiniteScroll.tsx

```diff
@@ -0,0 +1,108 @@
+    /** Components to render, should include entities/loading states/end message */
+    children: React.ReactNode;
+    /** Does the resource have more entities */
+    hasMore: boolean;
+    /** Are more entities being loaded */
+    isLoading: boolean;
+    /** Callback to load more entities */
+    onLoadMore: () => void;
+    /** React ref of the scroll container.
+     * Used to listen to scroll events and calculate onLoadMore calls.
+     * Set useWindow to true to use the window as scroll container.
+     */
+    scrollContainerNode?: HTMLElement;
+    /** Scroll threshold */
+    threshold?: number;
+    /** Throttle rate */
+    throttle?: number;
+    /** Set useWindow to true to use the window as scroll container. If set to true, will ignore scrollContainerRef. */
+    useWindow?: boolean;
+};
+
+function InfiniteScroll({
+    children,
+    hasMore,
+    isLoading,
+    onLoadMore,
+    scrollContainerNode,
+    threshold = 100,
+    throttle = 64,
+    useWindow = false,
+}: Props) {
+    const sentinelRef = React.createRef<HTMLDivElement>();
+
+    function getContainer(): EventTarget | undefined {
+        return useWindow ? window : scrollContainerNode;
+    }
+
+    React.useEffect(() => {
+        function onContainerScroll() {
+            if (isLoading || !hasMore) return;
+
+            if (sentinelRef.current == null) return;
+            const { top: sentinelTop } = sentinelRef.current.getBoundingClientRect();
+
+            if (useWindow) {
+                if (sentinelTop - window.innerHeight < threshold) {
+                    onLoadMore();
+                }
+            } else {
+                if (!scrollContainerNode) return;
+
+                const { bottom: containerBottom } = scrollContainerNode.getBoundingClientRect();
+                if (sentinelTop - containerBottom < threshold) {
+                    onLoadMore();
+                }
+            }
+        }
+        const scrollHandler = throttleFn(onContainerScroll, throttle);
+        const resizeHandler = throttleFn(onContainerScroll, throttle);
+
+        function addEventListeners() {
+            const container = getContainer();
+            if (container) {
+                container.addEventListener('scroll', scrollHandler);
+                container.addEventListener('resize', resizeHandler);
+            }
+        }
+
+        addEventListeners();
+
+        return function removeEventListeners() {
+            const container = getContainer();
+            if (container) {
+                container.removeEventListener('scroll', scrollHandler);
+                container.removeEventListener('resize', resizeHandler);
+            }
+        };
+    }, [
+        getContainer,
+        hasMore,
+        isLoading,
+        onLoadMore,
+        scrollContainerNode,
+        sentinelRef,
+        threshold,
+        throttle,
+        useWindow,
+    ]);
+
+    const sentinel = <div ref={sentinelRef} data-testid="sentinel" />;
```

Comment:
What is the reason to assign this to var instead of just declaring it in line 103?

Reply:
That was the original code, I am happy to change that